diff -r -u a/src/configure.in b/src/configure.in
--- a/src/configure.in	2008-02-01 12:39:54.000000000 +0200
+++ b/src/configure.in	2010-12-30 18:06:21.848661367 +0200
@@ -1,6 +1,8 @@
 dnl Process this file with autoconf to produce a configure script.
 
-AC_INIT(pl-wam.c)
+AC_PREREQ(2.61)
+
+AC_INIT(pl-wam.c, swi-prolog, 5.6.50)
 
 AC_CONFIG_HEADER(config.h)
 AC_SUBST(COFLAGS)
@@ -139,6 +141,9 @@
 	      configgmp="$enableval",
 	      configgmp=yes)
 
+AC_ARG_ENABLE(curses, [  --disable-curses	  Disable (n)curses(w) usage],
+	      configcurses="$enableval",
+	      configcurses=yes)
 
 dnl ****************************************************************
 dnl Setup PLBASE.  Need to expand prefix first as this isn't yet set. Yek
@@ -639,25 +644,32 @@
 
 
 AC_CHECK_LIB(m, main)
-AC_CHECK_LIB(ncursesw, main)
-if test ! "$ac_cv_lib_ncursesw_main" = "yes"; then
-  AC_CHECK_LIB(ncurses, main)
-  if test ! "$ac_cv_lib_ncurses_main" = "yes"; then
-    AC_CHECK_LIB(curses, main)
-    if test ! "$ac_cv_lib_curses_main" = "yes"; then
-      AC_CHECK_LIB(termcap, main)
-      if test "$ac_cv_lib_termcap_main" = "yes"; then
-	AC_MSG_CHECKING("whether -ltermcap needs -lucb")
-	AC_TRY_LINK([],
-		    [ tgetent();
-		    ],
-		    AC_MSG_RESULT(no)
-		    true,
-		    AC_MSG_RESULT(maybe ...)
-		    checkucb=yes)
+
+
+if test ! "$runtime" = "yes" -a "$configcurses" = "yes"; then
+  AC_CHECK_LIB(ncursesw, main)
+  if test ! "$ac_cv_lib_ncursesw_main" = "yes"; then
+    AC_CHECK_LIB(ncurses, main)
+    if test ! "$ac_cv_lib_ncurses_main" = "yes"; then
+      AC_CHECK_LIB(curses, main)
+      if test ! "$ac_cv_lib_curses_main" = "yes"; then
+        AC_CHECK_LIB(termcap, main)
+        if test "$ac_cv_lib_termcap_main" = "yes"; then
+  	  AC_MSG_CHECKING("whether -ltermcap needs -lucb")
+	  AC_TRY_LINK([],
+		      [ tgetent();
+		      ],
+		      AC_MSG_RESULT(no)
+		      true,
+		      AC_MSG_RESULT(maybe ...)
+		      checkucb=yes)
+        fi
       fi
     fi
   fi
+else
+  AC_MSG_CHECKING("curses support")
+  AC_MSG_RESULT("explicitly disabled")
 fi
 
 if test "$checkucb" = "yes"; then
