--- pl-5.6.50/src/configure.in	2010-12-15 22:02:56.636661749 +0200
+++ pl-5.6.50/src/configure.in	2010-12-15 22:08:52.721910027 +0200
@@ -1605,7 +1605,7 @@
 echo "	LDFLAGS=$LDFLAGS"
 echo "	LIBS=$LIBS"
 
-OutputFiles="Makefile rc/Makefile libtai/Makefile pl.1 pl.sh"
+OutputFiles="Makefile rc/Makefile libtai/Makefile pl.1 pl.sh pkconfig/swi-prolog.pc"
 
 dnl autoconf 2.5.3 doesn't handle relative srcdir properly.  We'll hack
 dnl around making this variable global.
diff -ruN a/pl-5.6.50/src/Makefile.in b/pl-5.6.50/src/Makefile.in
--- pl-5.6.50/src/Makefile.in	2008-02-01 12:39:54.000000000 +0200
+++ pl-5.6.50/src/Makefile.in	2010-12-15 22:33:41.715909870 +0200
@@ -384,6 +384,7 @@
 				$(MKDIR) $(DESTDIR)$(PLBASE)/$$d; \
 			fi; \
 		done
+		if [ ! -d $(DESTDIR)$(prefix)/lib/pkgconfig  ]; then $(MKDIR) $(DESTDIR)$(prefix)/lib/pkgconfig; fi
 
 
 iboot:		idirs
@@ -431,6 +432,7 @@
 			$(INSTALL_DATA) ../include/$$f $(DESTDIR)$(PLBASE)/include; \
 		    fi; \
 		done
+		$(INSTALL_DATA) pkconfig/swi-prolog.pc $(DESTDIR)$(prefix)/lib/pkgconfig
 
 iman::
 		-if [ ! -d $(DESTDIR)$(mandir) ]; then $(MKDIR) $(DESTDIR)$(mandir); fi
diff -ruN a/pl-5.6.50/src/pkconfig/swi-prolog.pc.in b/pl-5.6.50/src/pkconfig/swi-prolog.pc.in
--- pl-5.6.50/src/pkconfig/swi-prolog.pc.in	1970-01-01 02:00:00.000000000 +0200
+++ pl-5.6.50/src/pkconfig/swi-prolog.pc.in	2010-12-15 23:21:18.788909656 +0200
@@ -0,0 +1,11 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=${prefix}/lib/@PL@@RT@-@PLVERSION@/lib/@ARCH@
+includedir=${prefix}/lib/@PL@@RT@-@PLVERSION@/include
+base=${prefix}/lib/@PL@@RT@-@PLVERSION@
+
+Name: swi-prolog
+Description: SWI prolog interpreter runtime library.
+Version: @PLVERSION@
+Libs: -L${libdir} @LNLIBS@ -pthread 
+Cflags: -I${includedir}
